src/components/SignInForm/SignInForm.jsx

```diff
@@ -0,0 +1,170 @@
+import { useContext, useState, useCallback } from "react";
+import styles from "./SignInForm.module.scss";
+import { AuthContext } from "../../containers/AuthContext";
+
+const SignInForm = () => {
+  const authContext = useContext(AuthContext);
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const [errors, setErrors] = useState({});
+
+  const handleEyeButtonClick = (e) => {
+    e.preventDefault();
+    setIsPasswordVisible(!isPasswordVisible);
+  };
+
+  const handleEmailChange = (e) => {
+    setEmail(e.target.value);
+    setErrors((prevErrors) => ({ ...prevErrors, email: null }));
+  };
+
+  const handlePasswordChange = (e) => {
+    setPassword(e.target.value);
+    setErrors((prevErrors) => ({ ...prevErrors, password: null }));
+  };
+
+  const validateEmail = useCallback(() => {
+    const pattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (!pattern.test(email)) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        email: { valid: false, message: "Не правильна адреса" },
+      }));
+      return false;
+    }
+    setErrors((prevErrors) => ({
+      ...prevErrors,
+      email: { valid: true },
+    }));
+    return true;
+  }, [email]);
+
+  const validatePassword = useCallback(() => {
+    const pattern = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{9,}$/;
+    if (!pattern.test(password)) {
+      setErrors((prevErrors) => ({
+        ...prevErrors,
+        password: {
+          valid: false,
+          message:
+            "Не менше 8 буквенних та 1 числовий символи",
+        },
+      }));
+      return false;
+    }
+    setErrors((prevErrors) => ({
+      ...prevErrors,
+      password: { valid: true },
+    }));
+    return true;
+  }, [password]);
```

Comment:
Hi
can we try somehow using this approach for validation? 
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern

Reply:
please ignore for now,
probably not suitable for us at the moment ,
coz this will make validation much more aggressive 
but we want validate only when we leave the input
and we want to interact with other elements in the Dom. 
so let's keep our way.

Reply:
ok